backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java

```diff
@@ -0,0 +1,92 @@
+package com.redhat.cloud.notifications.templates;
+
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateInstance;
+
+// Name needs to be "Rhosak" to read templates from resources/templates/Rhosak
+public class Rhosak implements EmailTemplate {
+    private static final String SCHEDULED_UPGRADE = "scheduled-upgrade";
+    private static final String DISRUPTION = "disruption";
+    private static final String INSTANCE_CREATED = "instance-created";
+    private static final String INSTANCE_DELETED = "instance-deleted";
+    private static final String ACTION_REQUIRED = "action-required";
+
+    @Override
+    public TemplateInstance getTitle(String eventType, EmailSubscriptionType type) {
+        if (eventType.equals(SCHEDULED_UPGRADE)) {
+            return Templates.scheduledUpgradeTitle();
+        } else if (eventType.equals(DISRUPTION)) {
+            return Templates.serviceDisruptionTitle();
+        } else if (eventType.equals(INSTANCE_CREATED)) {
+            return Templates.instanceCreatedTitle();
+        } else if (eventType.equals(INSTANCE_DELETED)) {
+            return Templates.instanceDeletedTitle();
+        } else if (eventType.equals(ACTION_REQUIRED)) {
+            return Templates.actionRequiredTitle();
+        }
+
+        throw new UnsupportedOperationException(String.format(
+                "No email title template for RHOSAK event_type: %s and EmailSubscription: %s found.",
+                eventType, type
+        ));
+    }
+
+    @Override
+    public TemplateInstance getBody(String eventType, EmailSubscriptionType type) {
+        if (eventType.equals(SCHEDULED_UPGRADE)) {
+            return Templates.scheduledUpgradeBody();
+        } else if (eventType.equals(DISRUPTION)) {
+            return Templates.serviceDisruptionBody();
+        } else if (eventType.equals(INSTANCE_CREATED)) {
+            return Templates.instanceCreatedBody();
+        } else if (eventType.equals(INSTANCE_DELETED)) {
+            return Templates.instanceDeletedBody();
+        } else if (eventType.equals(ACTION_REQUIRED)) {
+            return Templates.actionRequiredBody();
+        }
```

Comment:
Similar as above

```suggestion
        if (type == EmailSubscriptionType.INSTANT) {
            if (eventType.equals(DISRUPTION)) {
                return Templates.serviceDisruptionBody();
            } else if (eventType.equals(INSTANCE_CREATED)) {
                return Templates.instanceCreatedBody();
            } else if (eventType.equals(INSTANCE_DELETED)) {
                return Templates.instanceDeletedBody();
            } else if (eventType.equals(ACTION_REQUIRED)) {
                return Templates.actionRequiredBody();
            }
        } else if (type == EmailSubscriptionType.DAILY) {
            return Templates.dailyBody();
        }
```

Reply:
Thanks for the review @josejulio the suggestion looks good to me other than the rename of `scheduledUpgradeBody` to `dailyBody` which to me is more generic and I'll rather it stays specific and more explicit to what the template is about rather than email frequency / type.

Reply:
Any name you pick is fine, I just want to clarify that in our current processing a single daily email is sent with all the aggregated event_types (You can choose what to include or not)

Reply:
Ah perfect thank you for pointing it out. I think what you are proposing as a name makes sense. I want to confirm first if this is what we expect i.e daily email of all event types rather than a daily email for each event type. 

/cc @akoserwal @davidffrench

Reply:
Both are very valid approaches to be honest, it comes down to personal preference. 

With the current architecture of the notification service where notifications are sent to all users in an organisation, this makes sense. However, in the future, once https://issues.redhat.com/browse/NOTIF-205 and https://issues.redhat.com/browse/NOTIF-206 have been completed, this will be more nuanced. If the logic for sending a single daily notification in this case will only include the notifications sent to the user during the day, this approach makes perfect sense.

Reply:
Perfect, thanks David. 

I have went ahead with the proposed change.

@josejulio do you want to have another look?